src/firebase-db.ts

```diff
@@ -222,7 +222,8 @@ export function createOrUpdateAnswer(answer: IExportableAnswerMetadata) {
       resource_url: portalData.resourceUrl,
       tool_id: portalData.toolId,
       tool_user_id: "anonymous",
-    };
+      platform_user_id: portalData.runKey
+    } as AnonymousRuntimeAnswerMetadata;
```

Comment:
Why is this casting necessary? Asking, as I think `as <type>` is being overused sometimes in our projects. It's usually better to declare variable as given type and then make an assigment, like:

`const something: AnonymousRuntimeAnswerMetadata = { ... }`

as then we have benefits from type checking.

Reply:
In this case postAnswer can be one of two different types.  I double checked and you are right that the cast with `as` doesn't do any type checking of the object being casted.
So added to `const` variables for each type so we get some actual type checking of the object literals here.

So we could just cast the first type as well on line 216 to make this more clear. However based on your comment it sounds like casting doesn't provide as much type checking 

We could just remove the `as` here. However because postAnswer can have two different types. It isn't staightforard it seems useful to say that when `portalData.type` is not "authenticated" that the type of the object should be. Because postAnswer can have two types depending on portalData.type. I think this `as AnonymousRuntimeAnswerMetadata`

Reply:
Looks good. I'd expect that:
`answerDocData: LTIRuntimeAnswerMetadata | AnonymousRuntimeAnswerMetadata;`
would let us use this variable in both if / else blocks, without this intermittent variable narrowed just to one type. But nothing bad in the current approach, maybe TS isn't smart enough yet.